eduhelx_jupyterlab_student/handlers.py

```diff
@@ -166,21 +209,40 @@ async def get(self):
                 cwd
             )
             # The cwd is the root in the frontend, so treat the path as such.
+            # NOTE: IMPORTANT: this field is NOT absolute on the server. It's only the absolute path for the webapp.
             assignment["absolute_directory_path"] = os.path.join("/", rel_assignment_path)
         value["assignments"] = assignments
 
-        # The student is in their repo, but we still need to check if they're actually in an assignment directory.
         current_assignment = student_repo.current_assignment
-        if current_assignment is not None:
-            submissions = await self.api.get_my_submissions(current_assignment["id"])
-            for submission in submissions:
-                submission["commit"] = get_commit_info(submission["commit_id"], path=student_repo.repo_root)
-            current_assignment["submissions"] = submissions
-
-            value["current_assignment"] = current_assignment
-            self.finish(json.dumps(value))
-        else:
-            self.finish(json.dumps(value))
+        # The student is in their repo, but we still need to check if they're actually in an assignment directory.
+        if current_assignment is None:
+            # If user is not in an assignment, we're done. Just leave current_assignment as None.
+            return json.dumps(value)
+        
+        submissions = await self.api.get_my_submissions(current_assignment["id"])
+        for submission in submissions:
+            submission["commit"] = get_commit_info(submission["commit_id"], path=student_repo.repo_root)
+        current_assignment["submissions"] = submissions
+        current_assignment["staged_changes"] = []
+        for modified_path in get_modified_paths(path=student_repo.repo_root):
+            full_modified_path = Path(student_repo.repo_root) / modified_path["path"]
+            abs_assn_path = Path(student_repo.repo_root) / assignment["directory_path"]
+            try:
+                path_relative_to_assn = full_modified_path.relative_to(abs_assn_path)
+                modified_path["path_from_repo"] = modified_path["path"]
+                modified_path["path_from_assn"] = str(path_relative_to_assn)
+                current_assignment["staged_changes"].append(modified_path)
```

Comment:
At this step `current_assignment.staged_changes` has `modified_path` property which holds two values -
1) path_from_repo
2) path_from_assn

just wanted to make sure, do I understand that logic correctly?

Reply:
Yes that's right